var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -52,7 +52,7 @@ class Silo(AutotoolsPackage):
     depends_on("automake", type="build", when="+shared")
     depends_on("libtool", type="build", when="+shared")
     depends_on("mpi", when="+mpi")
-    depends_on("hdf5@1.8", when="@:4.10+hdf5")
+    depends_on("hdf5@1.8:", when="@:4.10+hdf5")
     depends_on("hdf5@1.12:", when="@4.11:+hdf5")
```

Comment:
```suggestion
    with when("+hdf5"):
        depends_on("hdf5@1.8:")
        depends_on("hdf5@1.12:", when="@4.11:")
```

A bit easier to follow